src/Zynga/Framework/PgData/V1/PgModel/Reader.hh

```diff
@@ -286,6 +295,22 @@ class Reader implements ReaderInterface {
 
   }
 
+  private function getReaderOrWriterDatabase<TModelClass as PgRowInterface>(
+    bool $allowWriterOverride,
+    PgModelInterface $pgModel,
+    classname<TModelClass> $model,
+    PgWhereClauseInterface $where,
+  ): QueryableInterface {
+
+    if ($allowWriterOverride) {
```

Comment:
Actually why do you need allowWriterOverride? function doesWriterOverrideKeyExist already checks for the flag set in pgmodel and returns early

if (!$pgModel->allowWriterOverride()) {
        return false;
      }

Reply:
My thinking was to prevent having to do an extra call to memcache when we are accessing data based on the PK. Because when we are access data directly via the PK method, the data should not have been updated or recently added, so would not be effected by Replication lag.

Reply:
Maybe i missed something but you are only calling getReaderOrWriterDatabase in fetchResultSetFromDatabase so it won't be using the writer for getByPk()

Reply:
my bad i see fetchResultSetFromDatabase being called twice